File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    public List<List<Integer>> l;
    public List<Integer> ll;
    
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        l = new ArrayList<>();
        ll = new ArrayList<>();
        findSubsets(nums, 0);
        return l;
    }
    
    public void findSubsets(int[] nums, int i)
    {
        if (i == nums.length)
        {
            List<Integer> lll = new ArrayList<>(ll);
            Collections.sort(lll);
            if (!l.contains(lll))
            {
                l.add(new ArrayList<>(lll));
            }
            return;
        }
        
        ll.add(nums[i]);
        findSubsets(nums, i+1);
        
        ll.remove(ll.size() - 1);
        findSubsets(nums, i+1);
    }
}

File: 0090-subsets-ii/0090-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        temp = []
        nums.sort()
        def dp(i):
            nonlocal res, temp
            if i >= len(nums):
                res.append(temp.copy())
                return
            temp.append(nums[i])
            dp(i + 1)
            temp.pop()
            while i + 1 < len(nums) and nums[i] == nums[i+1]:
                i += 1
            dp(i + 1)
            return
        dp(0)
        return res